modules/memberships/class-jetpack-memberships.php

```diff
@@ -0,0 +1,283 @@
+			$this->get_plan_property_mapping()
+		);
+		return array_merge( $post_meta, array_values( $meta_keys ) );
+	}
+
+	public function return_meta( $map ) {
+		return $map['meta'];
+	}
+	/**
+	 * Callback that parses the membership purchase shortcode.
+	 *
+	 * @param array       $attrs - attributes in the shortcode. `id` here is the CPT id of the plan.
+	 * @param string|bool $content - needed for the callback.
+	 *
+	 * @return string|void
+	 */
+	public function render_button( $attrs ) {
+		Jetpack_Gutenberg::load_assets_as_required( self::$button_block_name);
+
+		if ( empty( $attrs['planId'] ) ) {
+			return;
+		}
+		$id = $attrs['planId'];
+		$product = get_post( $id );
+		if ( ! $product || is_wp_error( $product ) ) {
+			return;
+		}
+		if ( $product->post_type !== self::$post_type_plan || 'trash' === $product->post_status ) {
+			return;
+		}
+		$plan = self::product_post_to_array( $product );
+		$data = array(
+			'blog_id' => $this->get_blog_id(),
+			'id'	  => $id,
+			'button_label' => sprintf( __( '$%s Contribution', 'jetpack' ),$this->format_price( $plan ) ),
+			'powered_text' => __( 'Powered by WordPress.com', 'jetpack' ),
+		);
+
+		$classes = array(
+			'components-button',
+			'is-primary',
+			'is-button',
+			'wp-block-jetpack-' . self::$button_block_name,
+			self::$css_classname_prefix . '-' . $data['id'],
+		);
+		if ( isset( $attrs['className'] ) ) {
+			array_push( $classes, $attrs['className'] );
+		}
+		if ( isset( $attrs['submitButtonText'] ) ) {
+			$data['button_label'] = $attrs['submitButtonText'];
+		}
+		$button_styles = array();
+		if ( ! empty( $attrs['customBackgroundButtonColor'] ) ) {
+			array_push(
+				$button_styles,
+				sprintf(
+					'background-color: %s',
+					sanitize_hex_color( $attrs['customBackgroundButtonColor'] )
+				)
+			);
+		}
+		if ( ! empty( $attrs['customTextButtonColor'] ) ) {
+			array_push(
+				$button_styles,
+				sprintf(
+					'color: %s',
+					sanitize_hex_color( $attrs['customTextButtonColor'] )
+				)
+			);
+		}
+		$button_styles = implode( $button_styles, ';' );
+		add_thickbox();
+		return sprintf(
+			'<button data-blog-id="%d" data-powered-text="%s" data-plan-id="%d" class="%s" style="%s">%s</button>',
+			$data['blog_id'],
+			$data['powered_text'],
+			$data['id'],
+			implode( $classes, ' ' ),
+			$button_styles,
+			$data['button_label']
+		);
+	}
+
+	private function get_blog_id() {
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			return get_current_blog_id();
+		}
+
+		return Jetpack_Options::get_option( 'id' );
+	}
+
+	static function get_connected_account_id() {
+		return get_option( self::$connected_account_id_option_name );
+	}
+
+	/**
+	 * Formats the price.
+	 *
+	 * @param array $plan - array representing the plan.
+	 *
+	 * @return string
+	 */
+	private function format_price( $plan ) {
+		if ( $plan['formatted_price'] ) {
+			return $plan['formatted_price'];
+		}
+		return Jetpack_Simple_Payments::format_price( $plan['price'],$plan['currency'] );
```

Comment:
Re a method from another module feels little odd. @jeherve @kraftbj what's a good place for utility functions in Jetpack?

Reply:
Note that modules have "a contract" in Jetpack where they might be turned on or off and the code might not therefore exist. That said, I think Simple payments currently is _always_ loaded, but I wouldn't take that granted in terms of code-maintenance.

Reply:
We have `functions.global.php` for those things, for example. That file is not synced with WordPress.com right now, but we could copy it over.

Reply:
Nice, that would be good move. Like this it's quite prone to break unintentionally unless it's covered with tests.

Reply:
👍

Reply:
I removed the dependency on simple payments entirely.
This was just for providing a fallback in the PHP, but with recent UI changes it is very unlikely that it will be needed.
I ended uf providing a very generic fallback.